daemon/src/db/time_to_first_position.rs

```diff
@@ -0,0 +1,262 @@
+use anyhow::Result;
+use model::Identity;
+use sqlx::pool::PoolConnection;
+use sqlx::Sqlite;
+use time::OffsetDateTime;
+
+use crate::db::Connection;
+
+impl Connection {
+    /// Record the time at which we hear about a `taker_id` for the
+    /// first time.
+    ///
+    /// If we have already heard about the `taker_id`, we keep the
+    /// original `timestamp`.
+    pub async fn try_insert_first_seen(
+        &self,
+        taker_id: Identity,
+        timestamp: OffsetDateTime,
+    ) -> Result<()> {
+        let mut conn = self.inner.acquire().await?;
+
+        if !have_seen_before(&mut conn, taker_id).await? {
+            insert_first_seen(&mut conn, taker_id, timestamp).await?;
+        }
+
+        Ok(())
+    }
+
+    /// Record the time at which we hear about a taker with `taker_id`
+    /// opening a position for the first time.
+    ///
+    /// If we have already heard about the taker opening a position
+    /// before, we keep the original `timestamp`.
+    pub async fn try_insert_first_position(
+        &self,
+        taker_id: Identity,
+        timestamp: OffsetDateTime,
+    ) -> Result<()> {
+        let mut conn = self.inner.acquire().await?;
+
+        if !have_opened_a_position_before(&mut conn, taker_id).await? {
+            insert_first_position(&mut conn, taker_id, timestamp).await?;
+        }
+
+        Ok(())
+    }
+}
+
+async fn have_seen_before(conn: &mut PoolConnection<Sqlite>, taker_id: Identity) -> Result<bool> {
+    let first_seen_timestamp = load_first_seen_timestamp(conn, taker_id).await?;
+
+    Ok(first_seen_timestamp.is_some())
+}
+
+async fn load_first_seen_timestamp(
+    conn: &mut PoolConnection<Sqlite>,
+    taker_id: Identity,
+) -> Result<Option<OffsetDateTime>> {
+    let row = sqlx::query!(
+        r#"
+        SELECT
+            time_to_first_position.first_seen_timestamp
```

Comment:
```suggestion
            first_seen_timestamp
```

I don't think you need the table name here!

Reply:
yeah, in statements that only involve one table they are somewhat of an overkill.

Reply:
Yep, can do. I think we were trying to be consistent with always referencing the table. But it is overkill.